src/mono/mono/metadata/class.c

```diff
@@ -3037,18 +3037,22 @@ mono_image_init_name_cache (MonoImage *image)
 	const char *name;
 	const char *nspace;
 	guint32 visib, nspace_index;
-	GHashTable *name_cache2, *nspace_table, *the_name_cache;
+	dn_simdhash_u32_ptr_t *name_cache2;
+	dn_simdhash_string_ptr_t *nspace_table, *the_name_cache;
 
 	if (image->name_cache)
 		return;
 
-	the_name_cache = g_hash_table_new (g_str_hash, g_str_equal);
+	// TODO: Figure out a good initial capacity for this table by doing a scan,
```

Comment:
I think we can estimate a good guess for corelib (by the time this is first called, I think `mono_defaults.corlib` will be set already) and just set some kind of default for everything else.

Reply:
That makes sense. I suspect the current default (11-12 items) might be right for smaller images